addons/payment_worldline/models/payment_provider.py

```diff
@@ -0,0 +1,201 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import hashlib
+import hmac
+import logging
+import pprint
+from wsgiref.handlers import format_date_time
+
+import requests
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.fields import Datetime
+
+from odoo.addons.payment_worldline import const
+
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentProvider(models.Model):
+    _inherit = 'payment.provider'
+
+    code = fields.Selection(
+        selection_add=[('worldline', "Worldline")], ondelete={'worldline': 'set default'}
+    )
+    worldline_pspid = fields.Char(string="Worldline PSPID", required_if_provider='worldline')
+    worldline_api_key = fields.Char(string="Worldline API Key", required_if_provider='worldline')
+    worldline_api_secret = fields.Char(
+        string="Worldline API Secret", required_if_provider='worldline'
+    )
+    worldline_webhook_key = fields.Char(
+        string="Worldline Webhook Key", required_if_provider='worldline'
+    )
+    worldline_webhook_secret = fields.Char(
+        string="Worldline Webhook Secret", required_if_provider='worldline'
+    )
+
+    # === COMPUTE METHODS === #
+
+    def _compute_feature_support_fields(self):
+        """ Override of `payment` to enable additional features. """
+        super()._compute_feature_support_fields()
+        self.filtered(lambda p: p.code == 'worldline').update({
+            'support_tokenization': True,
+        })
+
+    # === BUSINESS METHODS === #
+
+    @api.model
+    def _setup_provider(self, code):
+        """ Override of `payment` to migrate Ogone tokens to Worldline.
+        Remove ogone from account payment method.
+        """
+        super()._setup_provider(code)
+        if code != 'worldline':
+            return
+
+        ogone_providers = self.env['payment.provider'].search([
+            ('code', '=', 'ogone'),
+            ('state', '!=', 'disabled'),
+        ])
+        if ogone_providers:
```

Comment:
```suggestion
        if not ogone_providers:
            return        
```
for better indentation 🙈